src/utils/ids.py

```diff
@@ -157,3 +194,41 @@ def handle_id_column(
 
     add_uuids_to_table(table, id_column)
     handle_existing_ids(table, table_name, id_mapping, id_column)
+
+
+def save_id_mapping(id_mapping: UUIDMapping, file_path: Path) -> None:
+    """Save ID mapping to a JSON file.
+
+    Args:
+        id_mapping: Dictionary mapping tuples to UUIDs
+        file_path: Path where to save the mapping
+    """
+    # Convert tuple keys to strings for JSON serialization
+    serializable_mapping = {json.dumps(key): value for key, value in id_mapping.items()}
```

Comment:
So the key is a json string, so when this gets dumped into a json, the keys will all be json lists with escaped quotes? I find this kind of horrifying, but I don't think you need to change it. I probably would have used a custom delimiter, like `|` or something, to join the keys into more readable strings. This is basically the same as that, but your custom delimiter is something like `\\", \\"`, I think?